crates/graphql/src/schema/query_root.rs

```diff
@@ -376,8 +381,17 @@ impl QueryRoot {
                 ),
             )
             .select(ListingColumns::default())
-            .load(&conn)
-            .context("Failed to load listings")?;
+            .into_boxed();
+
+        if let Some(limit) = limit {
+            query = query.limit(limit.into());
+        }
+
+        if let Some(offset) = offset {
+            query = query.offset(offset.into());
+        }
```

Comment:
Should validate that both or neither limit and offset are provided.

Reply:
I made the limit required and offset optional with default of 0